eclair-core/src/main/scala/fr/acinq/eclair/IAmReady.scala

```diff
@@ -0,0 +1,10 @@
+package fr.acinq.eclair
+
+/**
+ * Since actors are initialized asynchronously, and the initialization sometimes involves subscribing to an
+ * [[akka.event.EventStream]], we don't know when they are ready to process messages, especially in tests, which
+ * lead to race conditions.
+ * By making actors publish [[IAmReady]] on the same [[akka.event.EventStream]] they are subscribing to, we guarantee
+ * that if we receive [[IAmReady]] the actor has been initialized and its subscriptions have been taken into account.
+ */
+case class IAmReady(clazz: Class[_])
```

Comment:
I'm not a big fan of using `Class[_]` here, but I don't see a better identifier...

Reply:
Yes, the problem is for nested actors like `ChannelRelayer`, for which we don't have the `ActorRef`.